static/app/views/replays/replayTable.tsx

```diff
@@ -35,67 +35,82 @@ type RowProps = {
   showProjectColumn: boolean;
 };
 
-function ReplayTable({isFetching, replays, showProjectColumn, sort}: Props) {
+function SortableHeader({
+  fieldName,
+  label,
+  sort,
+}: {
+  fieldName: string;
+  label: string;
+  sort: Sort;
+}) {
   const location = useLocation<ReplayListLocationQuery>();
-  const organization = useOrganization();
-  const theme = useTheme();
-  const minWidthIsSmall = useMedia(`(min-width: ${theme.breakpoints.small})`);
-
-  const {pathname} = location;
 
   const arrowDirection = sort.kind === 'asc' ? 'up' : 'down';
   const sortArrow = <IconArrow color="gray300" size="xs" direction={arrowDirection} />;
 
+  return (
+    <SortLink
+      role="columnheader"
+      aria-sort={
+        sort.field.endsWith(fieldName)
+          ? sort.kind === 'asc'
+            ? 'ascending'
+            : 'descending'
+          : 'none'
+      }
+      to={{
+        pathname: location.pathname,
+        query: {
+          ...location.query,
+          sort: sort.kind === 'desc' ? fieldName : '-' + fieldName,
+        },
+      }}
+    >
+      {label} {sort.field === fieldName && sortArrow}
+    </SortLink>
+  );
+}
+
+function ReplayTable({isFetching, replays, showProjectColumn, sort}: Props) {
+  const organization = useOrganization();
+  const theme = useTheme();
+  const minWidthIsSmall = useMedia(`(min-width: ${theme.breakpoints.small})`);
+
   return (
     <StyledPanelTable
       isLoading={isFetching}
       isEmpty={replays?.length === 0}
       showProjectColumn={showProjectColumn}
       headers={[
         t('Session'),
-        showProjectColumn && minWidthIsSmall ? t('Project') : null,
-        <SortLink
+        showProjectColumn && minWidthIsSmall ? (
+          <SortableHeader
+            key="projectId"
+            sort={sort}
+            fieldName="projectId"
+            label={t('Project')}
+          />
+        ) : null,
```

Comment:
I would expect this to sort by slug

Reply:
I know :(

I talked to @cmanallen about it and the issue is that slug is mutable, so we're not comfortable puting it in the database as a simple column. I don't know what the effort is to things work and still be mutable at the data layer.
This is the same reason `projectSlug` isn't included in the api response.

I expect this will return things in a confusing order. But it could be useful to see things grouped together by project, especially if someone's filtered to see only two projects for example.
Also, it'll draw attention to the lack of `projectSlug` in the api response, which might help us find a solution to that. We're in the early days ,and Events include the field, so there might be something we can do.

Reply:
Yeah, I think elsewhere in the app we query in postgres for the slug